dnsaas/api/middlewares.py

```diff
@@ -0,0 +1,28 @@
+import re
+
+from django.core.urlresolvers import resolve
+from django.core.urlresolvers import reverse
+
+
+def get_api_version(request):
+    """Gets api version from header"""
+    value = request.META.get('HTTP_ACCEPT', '')
+    version = re.findall(r'v\d+', value, re.IGNORECASE)
+    version = version[0] if version else None
+    return version
+
+
+class VersionSwitch(object):
```

Comment:
could you describe (briefly) how does this work and what is the purpose of this middleware?